### DISCUSSION TITLE
course/starbook/word-detail

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#138] [General]
# course/starbook/word-detail

单词详情接口通过GET请求获取不在列表中的单词详细信息，包含例句、中文翻译、发音等字段。使用模糊路由匹配，并通过接口的逻辑层实现对数据库的查询操作。

https://goframe.org/course/starbook/word-detail

**Comment by nilsir** [comment]
大佬，教程上有个文件名写错了 api/words/v1/words.god   应为: api/words/v1/words.go，另外就是想请教下，这个路由参数是怎么解析给Id的呢，怎么设置为uint类型的

**Reply by gqcn** [reply]
路由参数会自动映射给请求数据结构，这个能力是框架提供的。

**Comment by jiumenyichang** [comment]
func (c *ControllerV1) Detail(ctx context.Context, req *v1.DetailReq) (res *v1.DetailRes, err error) {  
    uid, err := c.users.GetUid(ctx)     这块是不是有问题。c.什么呢

**Reply by c-v-c-v** [reply]
c *ControllerV1 当前的控制器，也就是this
